networkd/network_server.ml

```diff
@@ -1012,13 +1013,42 @@ module PVS_proxy = struct
     open S.PVS_proxy
 
     let path = ref "/opt/citrix/pvsproxy/socket/pvsproxy"
-
-	let do_call call =
-		try
-			Jsonrpc_client.with_rpc ~path:!path ~call ()
-		with e ->
-			error "Error when calling PVS proxy: %s" (Printexc.to_string e);
-			raise PVS_proxy_connection_error
+    
+    let do_call call =
+            let timebox ~timeout f =
+				let fd_mutex = Xapi_stdext_threads.Threadext.Mutex.create () in
```

Comment:
Is mutex amount limited in OCaml? If yes, we should move this mutex to the module level to make it a module member to avoid mutex leak.
I did some google search, the result shows the implementation in libc does not restrict the amount of mutex.

Reply:
It is better to have more fine grained locks, creating a new mutex should be as cheap as allocating some memory for it, the GC will take care of claiming it back.